main.js

```diff
@@ -11,28 +11,65 @@ function generateNumbers() {
     return allNumbers;
 }
 
-function displayNumbers(allNumbers) {
-    const numbersOnScreen = allNumbers.length;
+function displayNumbers(randomNumbers) {
+    const sortedNumbers = randomNumbers.slice(0).sort((a, b) => a - b);
+    const numbersOnScreen = randomNumbers.length;
+    let selectedOrder = [];
 
-     // using a for loop instead of forEach as the array.pop changes the length over each iteration
-     for (let i = 0; i < numbersOnScreen; i++) {
-        let randNumber = allNumbers.pop();
+    // using a for loop instead of forEach as the array.pop changes the length over each iteration
+    for (let i = 0; i < numbersOnScreen; i++) {
+        let randNumber = randomNumbers.pop();
         let number = document.createElement('div');
 
         number.textContent = randNumber;
         number.setAttribute('class', 'numberBox');
+
         document.querySelector('#play_area').appendChild(number);
     }
+
+    document.querySelectorAll('.numberBox').forEach((numberBox) => {
+        numberBox.addEventListener('click', () => {
+            const { textContent, classList } = numberBox;
+            const numberBoxValue = parseInt(textContent);
+
+            if (numberHasNotBeenClicked(classList)) {
+                if (isNextNumber(sortedNumbers, selectedOrder, numberBoxValue)) {
+                    selectedOrder.push(numberBoxValue);
+                    classList.add('clicked');
+                } else {
+                    document.querySelectorAll('.clicked').forEach((chosenNumber) => {
+                        chosenNumber.classList.remove('clicked');
+                        selectedOrder = [];
+                    });
+                }
+            }
+
+            if (allNumbersClicked(selectedOrder, sortedNumbers)) {
```

Comment:
this check happens if you click on a number thats been clicked before. it doesnt need to, so it can go inside the `numberHasNotBeenClicked` if, so theres less processing